src/components/media/MediasSelectModal.jsx

```diff
@@ -291,16 +338,290 @@ const MediasSelectModal = ({
+                        </Text>
+                        <Image
+                          alt={watch("altText")}
+                          src={
+                            watch("selectedMediaPath") ||
+                            `data:image/svg+xml;base64,${watch(
+                              "selectedMediaPath"
+                            )}`
+                          }
+                          fallback={
+                            <Flex
+                              background="base.canvas.alt"
+                              h="25rem"
+                              w="40rem"
+                              alignItems="center"
+                              justifyContent="center"
+                            >
+                              <Text color="base.content.medium">
+                                Your image will appear here.
+                              </Text>
+                            </Flex>
+                          }
+                          onLoad={(event) => {
+                            clearErrors("externalImageValidation")
+                            const { naturalWidth, naturalHeight } = event.target
+                            const maxSize = 5 * 1024 * 1024
+                            if (naturalHeight * naturalWidth > maxSize) {
+                              setError("externalImageValidation", {
+                                type: "custom",
+                                message: "The image size exceeds 5MB.",
+                              })
+                            }
+                          }}
+                          onError={() => {
+                            if (activeTab === 0) return
+                            if (!errors.selectedMediaPath)
+                              setError("externalImageValidation", {
+                                type: "custom",
+                                message:
+                                  "The provided URL does not point to a valid image.",
+                              })
+                          }}
+                        />
+                      </GridItem>
+                    </Grid>
+                  </TabPanel>
+                </TabPanels>
+              </Tabs>
+            </ModalBody>
+
+            <ModalFooter>
+              <Flex>
+                <LoadingButton
+                  alignSelf="flex-end"
+                  id="selectMedia"
+                  isDisabled={
+                    !watch("selectedMediaPath") ||
+                    !!errors.selectedMediaPath ||
+                    !!errors.altText ||
+                    !!errors.externalImageValidation
+                  }
+                  onClick={handleSubmit(async (data) => {
+                    if (activeTab === 0) {
+                      onProceed(data)
+                      return
+                    }
+                    onExternalProceed(data)
+                  })}
```

Comment:
in both use cases, `onProceed` and `onExternalProceed` appears to have no difference so we might not need this